pkg/syncer/metadatasyncer.go

```diff
@@ -248,6 +248,7 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
 		if err != nil {
 			return err
 		}
+		vCenter.Config.ReloadVCConfigForNewClient = true
```

Comment:
Why do we have to set ReloadVCConfigForNewClient during Init of Syncer ? Can you explain and add comments in the code

Reply:
```
// when ReloadVCConfigForNewClient is set to true it forces re-read config secret when
// new vc client needs to be created
ReloadVCConfigForNewClient bool
```
This flag was added and set to `true` for vanilla multi-vc only. 
This flag is been checked in virtualcenter.go/connect() method to re-read the config. 
I'm now setting it true for wcp controller and syncer as well.